### PR TITLE
Add a unit test for sets_ippermissions

### PR DESCRIPTION
**Pull request by nckturner**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>

/kind cleanup

**What this PR does / why we need it**:
Add a test for `sets_ippermissions_test.go` which is had a bug that is fixed in https://github.com/kubernetes/cloud-provider-aws/pull/250.

**Special notes for your reviewer**:
merge https://github.com/kubernetes/cloud-provider-aws/pull/250 first.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/providers/v1/sets_ippermissions_test.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+							VpcId:     aws.String("123"),
+						},
+					},
+					ToPort: aws.Int64(2),
+				},
+			),
+		},
+	}
+
+	for _, test := range tests {
```
Comment (M00nF1sh) [inline pkg/providers/v1/sets_ippermissions_test.go:150]: better to use subtests like below, use sub tests have multiple benefits: https://go.dev/blog/subtests
```
for _, test := range tests {
    t.Run(test.name, func(t *testing.T) {
        outputSet := test.inputSet.Ungroup()
	if !outputSet.Equal(test.expectedOutputSet) {
	     ....	
	}
   })
}
```

**Review by wongma7 — APPROVED** [review]

**Pull request merged**